src/client/mod.rs

```diff
@@ -31,8 +31,7 @@ pub struct Client {
 
 impl Client {
     pub fn connect(server: &str) -> ImapConnectFuture {
-        let addr = format!("{}:993", server);
-        let addr = addr.to_socket_addrs().unwrap().next().unwrap();
+        let addr = (server, 993).to_socket_addrs().unwrap().next().unwrap();
```

Comment:
This will panic depending on the value of `server`. `server` can be untrusted input. It in is our software: users enter an IMAP server address and it could be anything. So please return a `Result` instead.